File: packages/testing/statistical_tests/spot_checks/chi-square.test.R
#!/usr/bin/env Rscript

cat("\n📊 Chi-Square Test of Independence\n")
cat(paste(rep("-", 80), collapse = ""), "\n")

contingency_table <- matrix(c(10, 20, 30, 15, 25, 35), nrow = 2, byrow = TRUE)

# Chi-square test
result <- chisq.test(contingency_table)

cat("Contingency Table:\n")
print(contingency_table)
cat("\nChi-squared statistic:", result$statistic, "\n")
cat("df:", result$parameter, "\n")
cat("p-value:", result$p.value, "\n")

cat("\n", paste(rep("=", 80), collapse = ""), "\n")
cat("CHI-SQUARE TEST SPOT CHECK\n")
cat(paste(rep("=", 80), collapse = ""), "\n")

File: packages/testing/statistical_tests/spot_checks/chi-square.test.ts
import { stats } from "@tidy-ts/dataframe";

Deno.test("Chi-Square Test Spot Check", () => {
  console.log("\n📊 Chi-Square Test of Independence");
  console.log("-".repeat(80));

  const contingencyTable = [
    [10, 20, 30],
    [15, 25, 35],
  ];

  const result = stats.test.categorical.chiSquare({
    contingencyTable,
  });

  console.log("Contingency Table:", contingencyTable);
  console.log("Chi-squared statistic:", result.testStatistic.value);
  console.log("df:", result.degreesOfFreedom);
  console.log("p-value:", result.pValue);
});

console.log("\n" + "=".repeat(80));
console.log("CHI-SQUARE TEST SPOT CHECK");
console.log("=".repeat(80));
